zigpy/device.py

```diff
@@ -136,6 +153,39 @@ def schedule_initialize(self) -> asyncio.Task | None:
 
         return self._initialize_task
 
+    async def discover_ieee(self) -> EUI64:
+        self.info("Requesting IEEE address")
+
+        status, ieee, nwk, *_ = await self.zdo.IEEE_addr_req(
+            self.nwk, 0x00, 0x00, tries=2, delay=0.1
+        )
+
+        if status != zdo.types.Status.SUCCESS:
+            raise zigpy.exceptions.InvalidResponse(f"Requesting IEEE failed: {status}")
+
+        self.ieee = ieee
+        self.info("Discovered ieee address: %s", ieee)
+
+        return ieee
+
+    async def discover_nwk(self) -> NWK:
+        self.info("Requesting NWK address")
+
+        await zigpy.zdo.broadcast(
+            self.application, zdo.types.ZDOCmd.NWK_addr_req, 0, 0, self.ieee, 0x00, 0x00
+        )
+        status, ieee, nwk, *_ = await self.zdo.NWK_addr_req(
+            self.ieee, 0x00, 0x00, tries=2, delay=0.1, use_ieee=True
+        )
+
+        if status != zdo.types.Status.SUCCESS:
+            raise zigpy.exceptions.InvalidResponse(f"Requesting NWK failed: {status}")
+
+        self.nwk = nwk
```

Comment:
May need to check. The idea that this is part of the `Device.initialize()` method and once discover is complete it would be save into DB by `application.device_initialized()` method